triggering.go

```diff
@@ -55,6 +55,25 @@ func (dsp *DataStreamProcessor) triggerAtSpecificSamples(i int, NPresamples int,
 	return record
 }
 
+// firstPotentialTriggerFrame returns the first frame at which a trigger can be allowed in
+// the current segment. The isAuto flag says whether the triggers are auto-triggers (in which
+// case, the auto trigger delay is considered when determining the first frame).
+func (dsp *DataStreamProcessor) firstPotentialTriggerFrame(isAuto bool) int {
+	// dsp.LastTrigger stores the frame of the last trigger found by the most recent invocation of TriggerData
+	mindelay := dsp.NSamples
+	if isAuto {
+		autoDelaySamples := int(dsp.AutoDelay.Seconds()*dsp.SampleRate + 0.5)
+		if autoDelaySamples > mindelay {
+			mindelay = autoDelaySamples
+		}
+	}
+	nextPotentialTrig := int(dsp.LastTrigger-dsp.stream.firstFrameIndex) + mindelay
+	if nextPotentialTrig < dsp.NPresamples {
+		return dsp.NPresamples
+	}
+	return nextPotentialTrig
+}
+
```

Comment:
The new function `firstPotentialTriggerFrame` is here so that we can keep track of where triggering might be able to start in a new segment. The answer, of course, depends on whether we want to ask when _auto_ triggers might happen, or other types, because auto-triggers (potentially) have an extra delay beyond the usual no-overlap requirement.